src/vm/tests/datamaps.rs

```diff
@@ -56,6 +56,125 @@ fn test_simple_tea_shop() {
     assert_executes(expected, test1);
 }
 
+#[test]
+fn test_bound_tuple() {
+    let test =
+        "(define-map kv-store ((key int)) ((value int)))
+         (define (kv-add (key int) (value int))
+            (begin
+                (let ((my-tuple (tuple (key key))))
+                (insert-entry! kv-store (tuple (key key))
```

Comment:
I think `insert-entry!` should use `my-tuple` here to cover that insert case.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1007/commits/e476a33ae556a56062d316a6dc60fa0abe8135ee